### PR TITLE
Align token request with hosted Che

### LINKED ISSUE #16890: [Hosted Che] OpenShift Connector plugin does not work properly
### Is your task related to a problem? Please describe.
As part of https://github.com/redhat-developer/rh-che/issues/1832 we would like to enable the OpenShift Connector for every single workspace on Hosted Che. However, there are a couple of things that need to be fixed first:

- on Hosted Che console link is displayed via internal IP instead of full cluster link e.g. https://api.starter-us-east-2.openshift.com/ 
- user is logged in to the *-che namespace (should use username namespace)

image

### Describe the solution you'd like
OpenShift Connector plugin works properly on Hosted Che - console link is displayed correctly + user is logged to the `user` namespace by default.

### PR DESCRIPTION
**Pull request by vinokurig**
Check if plugin is running on Hosted Che, if so call specific Hosted Che token request. See: https://github.com/eclipse/che/issues/16890#issuecomment-636925592
fixes https://github.com/eclipse/che/issues/16890

### CODE REVIEW

File: src/che-openshift-authentication.ts [hunk 1]
```diff
@@ -11,10 +11,26 @@
...
+interface OsUserResponse {
+    data: Data[]
+}
 
 export async function start(context: theia.PluginContext) {
     const machineToken = process.env['CHE_MACHINE_TOKEN'];
     const isMultiUser = !!(machineToken && machineToken.length > 0);
+    const axiosInstance: AxiosInstance = axios;
+    const cheApi = process.env['CHE_API'];
+    const isHostedChe = cheApi && cheApi.indexOf('https://che.openshift.io/api') !== -1;
```
Comment (vitaliy-guliy) [inline src/che-openshift-authentication.ts:33]: Can we somehow avoid hardcoding of domain name here?

Comment (benoitf) [inline src/che-openshift-authentication.ts:33]: isn't it looking wired to hardcode an hosted platform there ?

File: package.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
     "@theia/plugin-packager": "latest",
     "rimraf": "2.6.2",
     "typescript-formatter": "7.2.2",
-    "typescript": "2.9.2"
+    "typescript": "2.9.2",
```
Comment (benoitf) [inline package.json]: is there a reason to not use same version of typescript of theia/che-theia ?

File: src/che-openshift-authentication.ts (hunk 1 above)
Comment (benoitf) [reply src/che-openshift-authentication.ts:33]: I mean, for example there is preview platform as well, etc, so maybe we need another way of saying it's an 'hosted che' ?

File: package.json (hunk 2 above)
Comment (vinokurig) [reply package.json]: set the version from `che-theia `

File: src/che-openshift-authentication.ts (hunk 1 above)
Comment (vinokurig) [reply src/che-openshift-authentication.ts:33]: Do you have any other ideas how to detect `Hosted Che`?

Comment (vinokurig) [reply src/che-openshift-authentication.ts:33]: We can make an API request but I think it is not a good solution.

**Comment by vinokurig** [comment]
@vitaliy-guliy @benoitf Do you have any objections to merge it?

File: src/che-openshift-authentication.ts [hunk 3]
```diff
@@ -86,7 +102,13 @@ export async function start(context: theia.PluginContext) {
     }
 
     function getServerUrl(): Promise<string> {
-        return new Promise<string>((resolve, reject) => {
+        return new Promise<string>(async (resolve, reject) => {
+            if (isHostedChe) {
+                const user = await che.user.getCurrentUser();
+                const osUserResponse = await axiosInstance.get<OsUserResponse>('https://api.openshift.io/api/users?filter[username]=' + user.name);
```
Comment (vitaliy-guliy) [inline src/che-openshift-authentication.ts:108]: I think it's possible to remove declarations of `isHostedChe` and `cheApi` and always ask the API.
```
if (process.env['CHE_API']) {
   const user = await che.user.getCurrentUser();
   const osUserResponse = await axiosInstance.get<OsUserResponse>(process.env['CHE_API'] + '/users?filter[username]=' + user.name);
   resolve(osUserResponse.data.data[0].attributes.cluster);
   return;
}
```
WDYT?

Comment (vinokurig) [reply src/che-openshift-authentication.ts:108]: This request is redundant in case when the plugin doesn't run on Hosted Che

**Comment by ericwill** [comment]
Let's merge this and make improvements for detecting Hosted Che later.

**Pull request merged**